system_contracts.md

```diff
@@ -0,0 +1,120 @@
+# System Contracts
+
+System contracts are interfaces defined as contracts, which are essential or
+recommended for an eWASM VM.
+
+An eWASM VM implementation may opt to implement these interfaces natively
+or to rely on implementations written in eWASM.
+
+Each of these contracts have a pre defined address and can be executed through
+regular contract invocations.
+
+## Sentinel Contract
+
+Address: `0x000000000000000000000000000000000000000a`
+
+Every newly deployed eWASM contract must be processed by the *Sentinel Contract*
+prior to including the code in the state. The sentinel will perform three very
+important processing steps:
+- Validate eWASM semantics
+- Inject metering code
+- Wrap the result in the deployer preamble (*see Appendix*)
+
+Input:
+- **variable length**: *eWASM contract code*
+
+Output:
+- **variable length**: *eWASM contract code*
+
+## EVM Transcompiler
+
+Address: `0x000000000000000000000000000000000000000b`
+
+This is optional. It is provided for pure eWASM VMs to support EVM1 contracts.
```

Comment:
yes but they will still need to implement the logic to run the EVM1 when a call goes to that address.